### ISSUE TITLE
I wrote a multi-gpu version based you on repo

### ISSUE THREAD

**Issue by hli2020** [multimodallearning/pytorch-mask-rcnn#19]
Hi @lasseha, I wrote a multi-gpu version based on your repo. Changed a lot on the whole code framework so I won't pull a request here. It is still in beta version. I will let you know (possibly open a brand new repo) when it is available (around the end of May I guess). Please have a (preliminary) look [here](https://github.com/hli2020/pytorch-mask-rcnn).

Thanks for you hard work! Right now the new multi-gpu version achieves around **36% mAP** on COCO, without any additional data augmentation (except the one you use: horizontal flip) or tricks. **Trained on 8 TItan X. Around 4 days. Not fully optimized.** I expect the training phase to be shorter. 

I have a few questions. Hopefully you can respond in a prompt manner.

- **COCO pretrain model**: how do you get this? there are weights on `mask.conv1.bias, size: torch.Size([256])`, `rpn`, `classifier`. Did you train a Mask-RCNN detection model and use it as pretrain? It is supposed to be classification model? 

- **batch norm learnable or not.** In line (here)[https://github.com/multimodallearning/pytorch-mask-rcnn/blob/master/model.py#L1461], you init BN layer as `requires_grad=False`, in this sense when the training is performed as stages (`heads`, `4+`, `all`), the BN layer is not trained at all, right? However, when you [set the optimizer](https://github.com/multimodallearning/pytorch-mask-rcnn/blob/master/model.py#L1786), the weight decay is explicitly removed in BN layer, why? Any experimental evidence?

Thanks!

**Comment by lasseha** [comment]
Hello @hli2020, that sounds great. Are your reported 36% mAP for the bounding box detections? What is your versions mAP for segmentation?

To answer your questions: 

- The pretrained model on COCO is converted from matterport's keras version. So i did not train this model at all.

- You are right. The batch norm layers are not trained, setting the weight decay explicitly in the optimizer is therefore not really necessary.

**Comment by liu6381810** [comment]
@hli2020 
Hi, I'm now trying to learn your multi-gpu repo, and in your repo, now there is no issue part
So I have to ask question here, sorry.
My question is that what the Dev means?  Maybe RoIAlign?
And I can't figure out what the config DEV.* mean, like DEV.SWITCH, BUFFER_SIZE
So could you please give me some explanations? Thanks!

**Comment by hli2020** [comment]
@liu6381810 
All the `DEV` options are my own investigation; they are irrelevant to Mask-RCNN framework. You can set `DEV.SWITCH=False`. My repo is still in preliminary version (lots of debug/profile statements); you can refer to a much better repo (pytorch with multiple gpus) [here](https://github.com/roytseng-tw/Detectron.pytorch). Strongly recommended.
